jazz_assets/components/validation.js

```diff
@@ -0,0 +1,311 @@
+        if (invalid_fields.length > 0) {
+            var message = "Following fields are invalid: " + invalid_fields.join(", ") + ". ";
+            reject({
+                "result": "inputError",
+                "message": message
+            });
+        } else {
+            resolve({
+                "result": "success",
+                "input": assets_data
+            });
+        }
+    });
+};
+
+function validateAllRequiredFields(assets_data, required_fields) {
+    logger.debug("Inside validateAllRequiredFields");
+    return new Promise((resolve, reject) => {
+        var missing_required_fields = _.difference(_.values(required_fields), _.keys(assets_data));
+        if (missing_required_fields.length > 0) {
+            var message = "Following field(s) are required - " + missing_required_fields.join(", ");
+            reject({
+                "result": "inputError",
+                "message": message
+            });
+        } else {
+            resolve({
+                "result": "success",
+                "input": assets_data
+            });
+        }
+    });
+};
+
+function validateInputFieldTypes(assets_data) {
+    logger.debug("Inside validateInputFieldTypes");
+    return new Promise((resolve, reject) => {
+        var invalid_fields = [];
+        Object.keys(assets_data).map((field) => {
+            if (assets_data[field]) {
+                if (!validateDataTypes(field, assets_data[field])) {
+                    invalid_fields.push(field);
+                }
+            }
+        })
+
+        if (invalid_fields.length > 0) {
+            var message = "The following field's value/type is not valid - " + invalid_fields.join(", ");
+            reject({
+                "result": "inputError",
+                "message": message
+            });
+        } else {
+            resolve({
+                "result": "success",
+                "input": "Input value is valid"
+            });
+        }
+    });
+};
+
+function validateDataTypes(field, prop_value) {
+    logger.debug("Inside validateDataTypes");
+    var fields_type = global.global_config.FIELD_DATA_TYPES;
+    var field_status = false;
+    Object.keys(fields_type).map((type) => {
+        if (field === type) {
+            if (fields_type[type] === 'String') {
+                if (prop_value && (typeof prop_value === 'string' || prop_value instanceof String)) {
+                    field_status = true;
+                }
+            } else if (fields_type[type] === 'Array') {
+                if (prop_value && prop_value.constructor === Array) {
+                    field_status = true;
+                }
+            }
+        }
+    });
+    return field_status;
+};
+
+function validateEnumValues(assets_data) {
+    logger.debug("Inside validateEnumValues");
+    return new Promise((resolve, reject) => {
+        var invalid_fields = [];
+        Object.keys(assets_data).map((field) => {
```

Comment:
same thing. return values which will be pushed to the array invalid_fields

Reply:
Updated